src/Microsoft.ML.Data/Transforms/HashTransform.cs

```diff
@@ -1258,5 +1294,63 @@ public override void Process()
             }
         }
     }
+
+    public sealed class HashConverter : IEstimator<HashConverterTransformer>
+    {
+        public const int NumBitsMin = 1;
+        public const int NumBitsLim = 32;
+
+        public static class Defaults
+        {
+            public const int HashBits = NumBitsLim - 1;
+            public const uint Seed = 314489979;
+            public const bool Ordered = false;
+            public const int InvertHash = 0;
+        }
+
+        private readonly IHost _host;
+        private readonly HashConverterTransformer.ColumnInfo[] _columns;
+
+        public static bool IsColumnTypeValid(ColumnType type)
+        {
+            return (type.ItemType.IsText || type.ItemType.IsKey || type.ItemType == NumberType.R4 || type.ItemType == NumberType.R8);
+        }
+        internal const string ExpectedColumnType = "Expected Text, Key, Single or Double item type";
+
+        public HashConverter(IHostEnvironment env, string name, string source = null,
```

Comment:
>string name, string source = null [](start = 51, length = 33)

lets follow the pattern and have `string inputColumn, string outputColumn = null`